canopen/network.py

```diff
@@ -197,10 +216,19 @@ def notify(self, can_id, data, timestamp):
         :param float timestamp:
             Timestamp of the message, preferably as a Unix timestamp
         """
-        if can_id in self.subscribers:
-            callback = self.subscribers[can_id]
-            callback(can_id, data, timestamp)
-        self.scanner.on_message_received(can_id)
+
+        # NMT commands is sent out with can_id = 0
+        if can_id == 0:
+            (cmd, node_id) = struct.unpack_from("<BB", data)
+
+            if node_id in self.nmt_cmd_subscribers:
```

Comment:
You should also add a case for when `node_id` is 0, then all NMT callbacks should be called as it is a broadcast message.

Reply:
Updated with Broadcast support